storage/database/sharded_database.go

```diff
@@ -151,102 +154,283 @@ func (pdb *shardedDB) Close() {
-}
-
-func (pdi *shardedDBIterator) Key() []byte {
-	// TODO-Klaytn implement this later.
-	return nil
-}
-
-func (pdi *shardedDBIterator) Value() []byte {
-	// TODO-Klaytn implement this later.
-	return nil
-}
-
-func (pdi *shardedDBIterator) Release() {
-	// TODO-Klaytn implement this later.
+func (it *shardedDBIterator) Error() error {
+	var err error
+	for i, iter := range it.iterators {
+		if iter.Error() != nil {
+			logger.Error("[shardedDBIterator] error from iterator",
+				"err", err, "shardNum", i, "key", it.key, "val", it.value)
+			err = iter.Error()
+		}
+	}
+	return err
+}
+
+func (it *shardedDBIterator) Key() []byte {
+	return it.key
+}
+
+func (it *shardedDBIterator) Value() []byte {
+	return it.value
+}
+
+func (it *shardedDBIterator) Release() {
+	it.cancel()
+}
+
+type entry struct {
+	key, val []byte
+}
+
+// shardedDBIterator iterates all items of each shardDB.
+// This is useful when you want to get items in serial.
+type shardedDBIteratorUnsorted struct {
+	shardedDBIterator
+}
+
+// NewIteratorUnsorted creates a iterator over the entire keyspace contained within
+// the key-value database.
+// If you want to get ordered items in serial, checkout shardedDB.NewIterator()
+// If you want to get items in parallel from channels, checkout shardedDB.NewChanIterator()
+func (db *shardedDB) NewIteratorUnsorted() Iterator {
+	return db.newIteratorUnsorted(func(db Database) Iterator { return db.NewIterator() })
+}
+
+// NewIteratorWithStartUnsorted creates a iterator over a subset of database content
+// starting at a particular initial key (or after, if it does not exist).
+func (db *shardedDB) NewIteratorWithStartUnsorted(start []byte) Iterator {
+	return db.newIteratorUnsorted(func(db Database) Iterator { return db.NewIteratorWithStart(start) })
+}
+
+// NewIteratorWithPrefixUnsorted creates a iterator over a subset of database content
+// with a particular key prefix.
+func (db *shardedDB) NewIteratorWithPrefixUnsorted(prefix []byte) Iterator {
+	return db.newIteratorUnsorted(func(db Database) Iterator { return db.NewIteratorWithPrefix(prefix) })
+}
+
+func (db *shardedDB) newIteratorUnsorted(newIterator func(Database) Iterator) Iterator {
+	it := &shardedDBIteratorUnsorted{
+		shardedDBIterator{
+			db.NewChanIterator(context.Background(), newIterator),
+			make(chan entry, shardedDBCombineChanSize),
+			nil, nil}}
+
+	go it.newCombineWorker()
+
+	return it
+}
+
+// newCombineWorker fetches any key/value from resultChs and put the data in resultCh
+func (it shardedDBIteratorUnsorted) newCombineWorker() {
+Iter:
+	for len(it.resultChs) != 0 {
+	chanIter:
+		for i, ch := range it.resultChs {
+			select {
+			case <-it.ctx.Done():
+				logger.Trace("[shardedDBIteratorUnsorted] combine worker ends due to ctx")
+				break Iter
+			case e, ok := <-ch:
+				if !ok { // channel is closed
+					it.resultChs = append(it.resultChs[:i], it.resultChs[i+1:]...)
+					break chanIter
+				} else {
+					it.resultCh <- e
+				}
+			default:
```

Comment:
Does this default case make infinity loop?
I think we should not use default case. Without this default, sleeping/waiting the channel is better.

So how about making each goroutine that pass items from each `ch` to `it.resultCh`?